src/Path.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Astrotomic\Path;
+
+use Astrotomic\Path\Posix\Path as PosixPath;
+use Astrotomic\Path\Win32\Path as Win32Path;
+
+class Path extends AbstractPath
+{
+    public static function isWindows(): bool
+    {
+        return (PHP_OS_FAMILY === 'Windows');
+    }
+
+    public static function posix(): PosixPath|PathContract
+    {
+        return new PosixPath();
+    }
+
+    public static function win32(): Win32Path|PathContract
+    {
+        return new Win32Path();
+    }
```

Comment:
I don't actually love that this is an instance of the object, nor that it's a new instance every time it's called.
However I think worrying about that right now might be a micro optimization in the long run.

The benefit of just returning the object is that it's the easiest way to provide autocompletion. It's also the easiest way to provide the fluent access to either Posix or Win32 apis on the fly.

Technically, we can use `::class` to simply return a string from both of these and get the same effect. However doing so looses the completion. Re-adding a doctype for the return could fix that though, however may limit the IDEs that get autocompletion from that.